packages/cli/src/ActiveExecutions.ts

```diff
@@ -74,9 +75,7 @@ export class ActiveExecutions {
 			}
 
 			executionId = await this.executionRepository.createNewExecution(fullExecutionData);
-			if (executionId === undefined) {
-				throw new ApplicationError('There was an issue assigning an execution id to the execution');
-			}
```

Comment:
this check has never made sense. the only reason the types here suggest that `executionId` could be missing is because we are reusing the variable. This code will never run, since if creating an execution fails, we'll have an unhandled error.

Reply:
Exactly. That's why I changed it to an `assert` so it satisfies the type system